abi/README.md

```diff
@@ -0,0 +1,51 @@
+# ABI Package
+
+## Overview
+The ABI package provides functionality for marshaling and unmarshaling actions without importing the codec package. It is designed to work across different language implementations.
+
+## ABI Format
+The ABI is defined in JSON format, as shown in the `abi.json` file:
+```json
+{
+    "actions": [
+        {
+            "id": 1,
+            "action": "MockObjectSingleNumber"
+        },
+        //...
+    ],
+    "types": [
+        {
+            "name": "MockObjectSingleNumber",
+            "fields": [
+                {
+                    "name": "Field1",
+                    "type": "uint16"
+                }
+            ]
+        },
+        //...
+}
+```
+
+The ABI consists of two main sections:
+- actions: A list of action definitions, each with an id and action name.
```

Comment:
Could we explain here that the action name must also be included as a type ie. they're coupled and we use the dictionary of types to interpret the actions?